mesos-client/src/main/scala/mesosphere/mesos/examples/UselessFramework.scala

```diff
@@ -0,0 +1,65 @@
+package mesosphere.mesos.examples
+
+import java.util.UUID
+
+import akka.actor.ActorSystem
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.{Sink, Source}
+import com.typesafe.scalalogging.StrictLogging
+import mesosphere.mesos.client.{MesosClient, StrictLoggingFlow}
+import mesosphere.mesos.conf.MesosClientConf
+import org.apache.mesos.v1.Protos.{Filters, FrameworkID, FrameworkInfo}
+import org.apache.mesos.v1.scheduler.Protos.Event
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+import scala.util.{Failure, Success}
+import collection.JavaConverters._
+
+/**
+  * Run Foo framework that:
+  *  - successfully subscribes
+  *  - declines all offers.
+  *
+  *  Not much, but shows the basic idea. Good to test against local mesos.
+  *
+  */
+object UselessFramework extends App with StrictLoggingFlow {
```

Comment:
we have a project for examples... should we move examples to that ?

Reply:
Man, I just posted the same. 😂 Good to know that we all converge to the same structure.

Reply:
Here is why it could make sense: the `usi/examples` folder is for examples _of the whole thing_ e.g. full-fledged frameworks capable of starting tasks on Mesos (at least that is my understanding of it). This `mesos-client` module is however usable on its own - it can't start anything but it can connect to Mesos.

Reply:
Yeah, this lives in this repo just to ease the initial development. In the future, this could be a separate repo and separate library used on its own - that's how it was designed from the beginning. Originally we planned to plug it in Marathon (which is still open possibility).

So I think considering that, examples on this level make sense and I like that @zen-dog did it this way

Reply:
I'd not ship it with the module though. how about making it a test target then? The Gradle project should not have a main.